### ISSUE TITLE
Changelog improvements

### ISSUE THREAD

**Issue by k3n** [ohmyzsh/ohmyzsh#11545]
### If the feature solves a problem you have, specify it here.

OMZ helpfully nudges me to update, and when I do, it lists a lot of things that were changed but it's a chore to get more details.

For instance, the latest update contained this line:

```
- 041c35f [amuse]                Add virtualenv support (#8987)
```

Sounds cool, but how does it work? It links the issue #, but now I have to open a browser, navigate to the project, then nav to its issues, and then lastly nav to #8987. Is it hard? No. Is it time-consuming? When there's 25 issues mentioned (like today), it can really add up!

That's part of the friction. The other part is updates like this:

```
 - cc73a92 [git]                  Add `gpod` alias
```

I understand that not all commits have an issue # to tie them to, but there's a git hash -- I can look at the code changes, but that requires roughly same amount of steps as above.

### Describe the proposed feature.

Add clickable links for each git hash & issue #. I'm making an assumption that most (if not all) of these refs are to the main project; if that's true, then this should be pretty straightforward. If not, then maybe alternatives would need to be considered.

>  - [041c35f](https://github.com/ohmyzsh/ohmyzsh/commit/041c35f) [amuse]                Add virtualenv support ([#8987](https://github.com/ohmyzsh/ohmyzsh/pull/8987))
>  - [cc73a92](https://github.com/ohmyzsh/ohmyzsh/commit/cc73a92) [git]                  Add `gpod` alias

### Describe alternatives you've considered

An alternative to providing this in the terminal would be to publish such and link it from the terminal, but I think having the details available in your terminal would be the easiest to work with.

### Additional context

I apologize if this is rehashing old decisions, I searched the issues (and online) but it seems the only way I can find details on the updated items is to do a ton of manual navigating myself. This seems like such an obvious omission that I can't but think it's intentional for some reason?

**Comment by mcornella** [comment]
Hi @k3n, thank you very much for the thorough feature request, it shows that you took the time to provide an actionable feedback and did the research.

To first answer your last point, this hasn't been discussed yet. To be fair, the only way to add links to both commits and issues on the terminal output is if the terminal emulator supports hyperlinks (i.e. [terminal hyperlinks](https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda) feature).

We are already testing for that feature on a variety of terminals in the installer and updater, so it isn't difficult to adapt the terminal changelog to add suport for that. If the terminal doesn't support this, it isn't feasible to add links as they'd be fully visible in the terminal output. 

The other option you sh, linking to a browser changelog, is only possible once we start using GitHub releases but that will come when we use a proper versioning process which is a long time coming.

So to summarize, I'll start working on adding hyperlinks to the output which is what can be shipped sooner. Do check if your terminal will support this feature though.

**Comment by k3n** [comment]
Using iTerm2 on Mac, links do indeed work. As a matter of fact, the current update text (which includes the changelog) includes clickable links to Discord and another link or 2.

This is great news, thank you!

**Comment by k3n** [comment]
This is A++++, thank you @mcornella! Saw it today and was excited :D

**Comment by mcornella** [comment]
This was all @GuySartorelli! I agree they did a great job!

**Comment by k3n** [comment]
Thank you @GuySartorelli!!
